OpenRA.Mods.Common/Warheads/TargetDamageWarhead.cs

```diff
@@ -36,20 +35,32 @@ protected override void DoImpact(WPos pos, Actor firedBy, WarheadArgs args)
 				if (!IsValidAgainst(victim, firedBy))
 					continue;
 
-				var closestActiveShape = victim.TraitsImplementing<HitShape>()
-					.Where(Exts.IsTraitEnabled)
-					.Select(s => (HitShape: s, Distance: s.DistanceFromEdge(victim, pos)))
-					.MinByOrDefault(s => s.Distance);
+				HitShape closestActiveShape = null;
+				var closestDistance = int.MaxValue;
+
+				// PERF: Avoid using TraitsImplementing<HitShape> that needs to find the actor in the trait dictionary.
+				foreach (var targetPos in victim.EnabledTargetablePositions)
+				{
+					if (targetPos is HitShape h)
```

Comment:
```suggestion
					if (targetPos is HitShape hitshape)
```
maybe? one letter variables are cryptic.